src/analysis.cpp

```diff
@@ -1311,40 +1311,60 @@ void ScatteringFunction::_sample() {
     for (int id : ids) { // loop over molecule names
         auto groups = spc.findMolecules(id);
         for (auto &g : groups) // loop over groups
-            if (usecom && !g.atomic)
+            if (use_com && !g.atomic)
                 p.push_back(g.cm);
             else
                 for (auto &i : g) // loop over particle index in group
                     p.push_back(i.pos);
     }
+
+    // zero-padded suffix to use with `save_after_sample`
+    std::string suffix = std::to_string(cnt);
+    suffix = std::string(7 - suffix.length(), '0') + suffix;
+
     switch (scheme) {
     case DEBYE:
         debye->sample(p, spc.geo.getVolume());
+        if (save_after_sample)
+            IO::write(filename + "." + suffix, debye->getIntensity());
         break;
-    case EXPLICIT:
-        explicit_average->sample(p, spc.geo.getLength().x());
+    case EXPLICIT_PBC:
+        explicit_average_pbc->sample(p, spc.geo.getLength().x());
+        if (save_after_sample)
+            IO::write(filename + "." + suffix, explicit_average_pbc->getSampling());
+        break;
+    case EXPLICIT_IPBC:
+        explicit_average_ipbc->sample(p, spc.geo.getLength().x());
+        if (save_after_sample)
+            IO::write(filename + "." + suffix, explicit_average_ipbc->getSampling());
         break;
     }
 }
 void ScatteringFunction::_to_json(json &j) const {
-    j = {{"molecules", names}, {"com", usecom}};
+    j = {{"molecules", names}, {"com", use_com}};
     switch (scheme) {
     case DEBYE:
         j["scheme"] = "debye";
-        j["qmin"] = debye->qmin;
-        j["qmax"] = debye->qmax;
+        std::tie(j["qmin"], j["qmax"], std::ignore) = debye->getQMeshParameters();
```

Comment:
In c++17, this should be valid: `auto [j["qmin"], j["qmax"], std::ignore) = debye->getQMeshParameters();]`

Reply:
The structure binding does not like array-like identifiers in the list. Kept.